crates/noirc_evaluator/src/ssa/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -441,17 +450,16 @@ impl AcirContext {
     /// Adds a new Variable to context whose value will
     /// be constrained to be the addition of `lhs` and `rhs`
     pub(crate) fn add_var(&mut self, lhs: AcirVar, rhs: AcirVar) -> Result<AcirVar, RuntimeError> {
-        let lhs_data = &self.vars[&lhs];
-        let rhs_data = &self.vars[&rhs];
-        let result_data = if let (AcirVarData::Const(lhs_const), AcirVarData::Const(rhs_const)) =
-            (lhs_data, rhs_data)
-        {
-            AcirVarData::Const(*lhs_const + *rhs_const)
-        } else {
-            let sum = lhs_data.to_expression().as_ref() + rhs_data.to_expression().as_ref();
-            AcirVarData::Expr(sum)
-        };
-        Ok(self.add_data(result_data))
+        let lhs_expr = self.var_to_expression(lhs)?;
+        let rhs_expr = self.var_to_expression(rhs)?;
+        let sum_expr = &lhs_expr + &rhs_expr;
+
+        if let Some(constant) = sum_expr.to_const() {
```

Comment:
We can remove this if statement as the constant case is handled by `AcirVarData::from`